### ISSUE TITLE
Left UI Shell: The responsive menu is not dismissable with Esc

### ISSUE THREAD

**Issue by tay1orjones** [carbon-design-system/carbon#13414]
From the parent issue:

> Problem 2: The responsive menu is not dismissable with Esc
> When the hamburger menu is activated, it becomes an X, which can be activated to close the left nav. This is an awkward implementation of a close function, which really doesn't work well with this disclosure pattern. There is no good reason not to cause the left nav to be dismissed on pressing Esc, and there are several good reasons to do it:
> 
> It allows the user to dismiss the temporarily displayed left nav from anywhere inside the menu
> Because the focus does not immediately go into the revealed left nav, but instead traverses the site, it is even more difficult for a keyboard user to reach the X to dismiss the nav
> As discussed in the next problem, because the main content is partially obscured with a light-box effect, the inability for the keyboard user to dismiss the left panel, either through Esc or by tabbing out of the left nav, becomes more problematic.
